services/interfaces/state_transition_tracker_interface.py

```
@@ -0,0 +1,44 @@
+        """
+        Record a state transition.
+
+        Args:
+            state: The processing state name
+            step_description: Human-readable description of the step
+            timestamp: When the transition occurred (must be provided)
+        """
+        pass
+
+    @abstractmethod
+    def get_transitions(self) -> List['StateTransition']:
```

Ruff (0.14.7)

Undefined name `StateTransition`

(F821)